### PR TITLE
BubbleSort Implementation

### PR DESCRIPTION
**Pull request by kunmi02**
This repository is a bubble sort algorithm implementation in ruby. The algorithm was implemented using yield which is wrapped up in a` bubble_sort_by` method and it was also implemented using the normal algorithm, which was wrapped up in  `bubble_sort` method.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,41 @@
...
+
+
+
+## Author
+
+👤 Suyash Fowdar
+Github: [@Krishnzzz](https://github.com/krishnzzz)
+
+👤 Azeez Akinkunmi
+Github: [@kunmi02](https://github.com/kunmi02)
```
Comment (descholar-ceo) [inline README.md]: [Optional] You can add more social network links such as (linkedin and twitter) if you have them

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      next unless !arr[index + 1].nil? && (arr[index + 1] < arr[index])
+
+      a = arr[index]
+      b = arr[index + 1]
+      arr[index + 1] = a
+      arr[index] = b
+      done = false
+    end
+  end
+  puts(arr)
```
Comment (descholar-ceo) [inline app.rb]: [Required] Don't use puts here, return that `arr` instead

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      a = arr[index]
+      b = arr[index + 1]
+      next unless yield(a, b).positive?
+
+      arr[index + 1] = a
+      arr[index] = b
+      done = false
+    end
+  end
+  puts(arr)
```
Comment (descholar-ceo) [inline app.rb]: [Required] Don't use puts here, return that arr instead

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Change Required :vertical_traffic_light: :warning: 
## Hello @kunmi02 and @krishnzzz  :handshake: :handshake: :handshake: 
:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
## Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes]
- [x] Don't use `puts` in your codes, instead you should return what you are printing on the console.
Re-read these project requirements [Screenshot 1](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sort.png) and [Screenshot 2](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) they are asking you to return something not to print it on screen.
- [x] Implement all changes I suggested in inline comments in your codes

# [Optional changes]
- [ ] You can Change your branchname, from `Sort&SortBy-feature` to something like `Sort-SortBy-feature` because checking out that branch with the amp symbol, didn't work on my local environment, I had to surround it with quotes to make it work, so for the sake of simplicity, you can rename it to make it very easy
- [ ] Your commit messages should be in imperative mode or present tense, _i.e.:_ you should have written `Fix linter issues` instead of writing `fixed linter issues`, refer [to this article for more details on how to write a good && professional commit message](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

If anything seems to be unclear to you, don't hesitate to contact me for more clarification slack: `@Mugirase Emmanuel` | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Review by layzrd — APPROVED** [review]
### Status: Approved 🙌 🎇
Hi @krishnzzz and @kunmi02 , Congratulations 🎊 👏, you did very well on this project.
 
Now, you can move ahead.

Happy coding 👨‍💻
[Abdulaziz Redi Ali ](https://azizali.ml/) | [Reach me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

**Pull request merged**